metrics/github/cli.py

```diff
@@ -50,22 +50,24 @@ def is_open(pr, dt):
 
         return (closed - opened) >= threshold
 
-    with TimescaleDBWriter(GitHubPullRequests) as writer:
-        for monday in mondays:
-            dt = datetime.combine(monday, time(), tzinfo=UTC)
-            prs_open = [pr for pr in prs if is_open(pr, dt)]
-            prs_open = drop_archived_prs(prs_open, monday)
+    for monday in mondays:
+        dt = datetime.combine(monday, time(), tzinfo=UTC)
+        prs_open = [pr for pr in prs if is_open(pr, dt)]
+        prs_open = drop_archived_prs(prs_open, monday)
 
-            name = f"queue_older_than_{days_threshold}_days"
+        name = f"queue_older_than_{days_threshold}_days"
 
-            log.info(
-                "%s | %s | Processing %s PRs open at %s",
-                name,
-                org,
-                len(prs_open),
-                dt,
-            )
-            process_prs(writer, prs_open, monday, name=name)
+        log.info(
+            "%s | %s | Processing %s PRs open at %s",
+            name,
+            org,
+            len(prs_open),
+            dt,
+        )
+        processed_prs = list(iter_prs(prs_open, monday, name))
```

Comment:
I don't think this can be right. You're throwing away the PRs for all but the last Monday, aren't you? Must be missing a test case in the test add later.

Reply:
oh yes, this is clearly wrong.  I thought I could pull the writing out of the loop, clearly not this way!  This gets fixed when the writing is pulled out of this function entirely and done in the calling function.